networking/gcp/vpc/main.tf

```diff
@@ -0,0 +1,121 @@
+# Configuration in the current provider
+data "google_client_config" "current" {}
+
+locals {
+  enable_external_access = length(var.public_subnets) > 0
+}
+
+# VPC
+resource "google_compute_network" "vpc" {
+  name                                      = var.name
+  description                               = "VPC deployment of name ${var.name}"
+  auto_create_subnetworks                   = var.auto_create_subnetworks
+  routing_mode                              = var.routing_mode
+  mtu                                       = var.mtu
+  enable_ula_internal_ipv6                  = var.enable_ula_internal_ipv6
+  internal_ipv6_range                       = can(coalesce(var.enable_ula_internal_ipv6)) ? var.internal_ipv6_range : null
+  network_firewall_policy_enforcement_order = var.network_firewall_policy_enforcement_order
+  project                                   = data.google_client_config.current.project
+  delete_default_routes_on_create           = var.delete_default_routes_on_create
+}
+
+# Shared VPC host project
+resource "google_compute_shared_vpc_host_project" "shared" {
+  count      = var.shared_vpc_host_project ? 1 : 0
+  project    = data.google_client_config.current.project
+  depends_on = [google_compute_network.vpc]
+}
```

Comment:
I am not sure why this is here.

Reply:
Optional if we need create a shared VCP host project

Reply:
Despite its name, shared VPC is not really about VPC. More precisely, you must enable the project to be a host project (the purpose of this resource), and you must give the permissions to use the VPC (or just subnetwork if you want) to service accounts outside this project. The former must be done only once, even if you create multiple VPCs in the project and is independent from the actual VPC you have created.

Reply:
OK